### DISCUSSION TITLE
Allow to provide "soft constraint"

### DISCUSSION THREAD

**Discussion by nbahar-intel** [python-constraint/python-constraint#87] [Ideas]
It'd be nice to have an option to provide a weight to a constraint. 
When there is no solution, the constraints with the smallest weight can be dropped (or ignored) in order to achieve a solution.

Would be also useful when wanting to provide a default value to a variable, without enforcing it when it's not able to assign that value due to constraints.

**Comment by fjwillemsen** [comment]
These seem like useful features indeed, however, it would require a lot of changes internally. I'll convert this to a discussion, if more people feel like this feature is needed we can look into it.
